qunicorn_core/core/job_service.py

```diff
@@ -101,10 +104,20 @@ def send_job_to_pilot():
     raise NotImplementedError
 
 
-def cancel_job_by_id(job_id):
+def cancel_job_by_id(job_id, token):
```

Comment:
Am besten auch in den job_manager_service moven, dann werden die Pilots nur vom job_manager_service und pilot_manager_service aufgerufen.